language/compiler/bytecode_source_map/src/disassembler.rs

```diff
@@ -111,17 +259,198 @@ impl<Location: Clone + Eq + Default> Disassembler<Location> {
                     format_err!(
-                        "Type parameter index out of bounds while disassembling type signature"
+                        "Type parameter index {} out of bounds while disassembling type signature",
+                        ty_param_index
                     )
                 })?
                 .0
                 .to_string(),
         })
     }
 
-    //***************************************************************************
-    // Disassemblers
-    //***************************************************************************
+    fn disassemble_instruction(
+        &self,
+        instruction: &Bytecode,
+        locals_sigs: &LocalsSignature,
+        function_source_map: &FunctionSourceMap<Location>,
+    ) -> Result<String> {
+        match instruction {
+            Bytecode::LdAddr(address_idx) => {
+                let address = self.source_mapper.bytecode.address_at(*address_idx);
+                Ok(format!("LdAddr({})", address))
+            }
+            Bytecode::LdByteArray(byte_array_idx) => {
+                let bytearray = self.source_mapper.bytecode.byte_array_at(*byte_array_idx);
+                Ok(format!("LdByteArray({:?})", bytearray))
+            }
+            Bytecode::LdStr(string_idx) => {
+                let string = self.source_mapper.bytecode.user_string_at(*string_idx);
+                Ok(format!("LdStr({})", string.as_str()))
+            }
+            Bytecode::CopyLoc(local_idx) => {
+                let (name, ty) =
+                    self.name_for_local(*local_idx as u64, locals_sigs, function_source_map)?;
+                Ok(format!("CopyLoc({}, {}: {})", local_idx, name, ty))
+            }
+            Bytecode::MoveLoc(local_idx) => {
+                let (name, ty) =
+                    self.name_for_local(*local_idx as u64, locals_sigs, function_source_map)?;
+                Ok(format!("MoveLoc({}, {}: {})", local_idx, name, ty))
+            }
+            Bytecode::StLoc(local_idx) => {
+                let (name, ty) =
+                    self.name_for_local(*local_idx as u64, locals_sigs, function_source_map)?;
+                Ok(format!("StLoc({}, {}: {})", local_idx, name, ty))
+            }
+            Bytecode::MutBorrowLoc(local_idx) => {
+                let (name, ty) =
+                    self.name_for_local(*local_idx as u64, locals_sigs, function_source_map)?;
+                Ok(format!("StLoc({}, {}: {})", local_idx, name, ty))
+            }
+            Bytecode::ImmBorrowLoc(local_idx) => {
+                let (name, ty) =
+                    self.name_for_local(*local_idx as u64, locals_sigs, function_source_map)?;
+                Ok(format!("ImmBorrowLoc({}, {}: {})", local_idx, name, ty))
+            }
+            Bytecode::MutBorrowField(field_idx) => {
+                let (name, ty) = self.name_for_field(*field_idx)?;
+                Ok(format!("MutBorrowField({}, {}: {})", field_idx, name, ty))
```

Comment:
Similar question here about whether we want to print the types.

Reply:
I think printing the type here is really nice/helpful. For locals there's much more of a case to be made for not printing the types since they are readily determinable, whereas for fields I feel it's significantly harder. So I would like to print the type info here unless you or @dariorussi have strong opinions against.